modules/phase_field/include/materials/GrandPotentialSinteringMaterial.h

```diff
@@ -111,6 +111,17 @@ class GrandPotentialSinteringMaterial : public DerivativeMaterialInterface<Mater
   /// interface profile coefficient
   MaterialProperty<Real> & _gamma;
 
+  /// Body Force for mass conservation in conc and chempot coupling
+  MaterialProperty<Real> & _bodyforcev;
+  MaterialProperty<Real> & _bodyforces;
+
+  /// MatReaction Force for mass conservation in conc and chempot coupling
+  MaterialProperty<Real> & _matreactv;
+  MaterialProperty<Real> & _matreacts;
+  
+  /// mobility between c and w coupling
+  MaterialProperty<Real> & _mob_conc_w;
```

Comment:
This is just a material property that is going to be `-1` everywhere, I am hoping we can define it right in the action rather than declaring it and computing here and passing in to the action. Please see further comments below